src/GafferScene/Preview/RendererAlgo.cpp

```diff
@@ -718,6 +721,125 @@ void outputObjects( const ScenePlug *scene, const IECore::CompoundObject *global
+		{
+			resolution = resolutionData->readable();
+		}
+
+		if( const FloatData *resolutionMultiplierData = globals->member<FloatData>( "option:render:resolutionMultiplier" ) )
+		{
+			resolution.x = int((float)resolution.x * resolutionMultiplierData->readable());
+			resolution.y = int((float)resolution.y * resolutionMultiplierData->readable());
+		}
+
+		const FloatData *pixelAspectRatioData = globals->member<FloatData>( "option:render:pixelAspectRatio" );
+		if( pixelAspectRatioData )
+		{
+			camera->parameters()["pixelAspectRatio"] = pixelAspectRatioData->copy();
+		}
+	}
+
+	camera->parameters()["resolution"] = new V2iData( resolution );
+
+	// calculate an appropriate screen window
+
+	camera->addStandardParameters();
+
+	// apply overscan
+
+
+	Box2i renderRegion( V2i( 0 ), resolution - V2i( 1 ) );
+
+	const BoolData *overscanData = globals->member<BoolData>( "option:render:overscan" );
+	if( overscanData && overscanData->readable() )
+	{
+		// get offsets for each corner of image (as a multiplier of the image width)
+		V2f minOffset( 0.0 ), maxOffset( 0.0 );
+		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanLeft" ) )
+		{
+			minOffset.x = overscanValueData->readable();
+		}
+		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanRight" ) )
+		{
+			maxOffset.x = overscanValueData->readable();
+		}
+		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanBottom" ) )
+		{
+			minOffset.y = overscanValueData->readable();
+		}
+		if( const FloatData *overscanValueData = globals->member<FloatData>( "option:render:overscanTop" ) )
+		{
+			maxOffset.y = overscanValueData->readable();
+		}
+
+		// convert those offsets into pixel values
+		renderRegion.min -= V2i(
+			int(minOffset.x * (float)resolution.x),
+			int(minOffset.y * (float)resolution.y)
+		);
+
+		renderRegion.max += V2i(
+			int(maxOffset.x * (float)resolution.x),
+			int(maxOffset.y * (float)resolution.y)
+		);
+	}
+
+	
+	const Box2fData *cropWindowData = globals->member<Box2fData>( "option:render:cropWindow" );
+	if( cropWindowData )
+	{
+		const Box2f &cropWindow = cropWindowData->readable();
+		Box2i cropRegion(
+			V2i( (int)( round( resolution.x * cropWindow.min.x ) ),
+			     (int)( round( resolution.y * cropWindow.min.y ) ) ),
+			V2i( (int)( round( resolution.x * cropWindow.max.x ) ) - 1,
+			     (int)( round( resolution.y * cropWindow.max.y ) ) - 1 ) );
+
+		renderRegion.min.x = std::max( renderRegion.min.x, cropRegion.min.x );
+		renderRegion.max.x = std::min( renderRegion.max.x, cropRegion.max.x );
+		renderRegion.min.y = std::max( renderRegion.min.y, cropRegion.min.y );
+		renderRegion.max.y = std::min( renderRegion.max.y, cropRegion.max.y );
+			
+	}
+
+	if( renderRegion.isEmpty() )
+	{
+		// OpenEXR does not permit empty data windows, so it is unlikely that any renderer
```

Comment:
Out of interest, what does Arnold do with an empty render region right now? It seems a bit odd to pander to OpenEXR's deficiencies in a function which is just about cameras, when OpenEXR might not even be in use. I think my preference might be to apply the workaround in the renderer backends if necessary.